src/pybind/mgr/dashboard/conftest.py

```diff
@@ -24,3 +24,16 @@ def pytest_configure(config):
         'rbd': Mock(),
         'cephfs': Mock(),
     })
+
+    # we need the following patches to fix the issue of multiple inheritance when
+    # one of the base classes is being mocked.
+    # Error example:
+    # TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) \
+    # subclass of the metaclasses of all its bases
+    class _BaseMgrModule:
+        pass
+
+    patcher = patch("ceph_module.BaseMgrStandbyModule", new=_BaseMgrModule)
+    patcher.start()
+    patcher = patch("ceph_module.BaseMgrModule", new=_BaseMgrModule)
+    patcher.start()
```

Comment:
Ha! I had a similar problem when fetching all mon command definitions in the mgr modules. I came up with a very similar solution

```python
from unittest import mock

class M(object):
    def __init__(self, *args):
        self._ceph_get_version = mock.Mock()

class B(object):
    def __init__(self, *args):
        self._ceph_get_version = mock.Mock()

def main():
    cm = mock.Mock()
    cm.BaseMgrModule = M
    cm.BaseMgrStandbyModule = B
    sys.modules['ceph_module'] = cm
    sys.modules['OpenSSL'] = mock.Mock()
    
    ...    

if __name__ == '__main__':
  main()
```